centreon/src/Core/Media/Application/UseCase/FindMedias/FindMedias.php

```diff
@@ -0,0 +1,101 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types = 1);
+
+namespace Core\Media\Application\UseCase\FindMedias;
+
+use Centreon\Domain\Contact\Contact;
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Centreon\Domain\Log\LoggerTrait;
+use Centreon\Domain\RequestParameters\Interfaces\RequestParametersInterface;
+use Centreon\Infrastructure\RequestParameters\RequestParametersTranslatorException;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Application\Common\UseCase\ForbiddenResponse;
+use Core\Media\Application\Exception\MediaException;
+use Core\Media\Application\Repository\ReadMediaRepositoryInterface;
+use Core\Media\Domain\Model\Media;
+
+final class FindMedias
+{
+    use LoggerTrait;
+
+    public function __construct(
+        readonly private RequestParametersInterface $requestParameters,
+        readonly private ReadMediaRepositoryInterface $readMediaRepository,
+        readonly private ContactInterface $user,
+    ) {
+    }
+
+    /**
+     * @param FindMediasPresenterInterface $presenter
+     */
+    public function __invoke(FindMediasPresenterInterface $presenter): void
+    {
+        try {
+            $this->info('Find medias', ['user' => $this->user->getId()]);
+            if (! $this->canAccessToListing()) {
+                $this->error(
+                    "User doesn't have sufficient rights to list medias",
+                    ['user_id' => $this->user->getId()]
+                );
+                $presenter->presentResponse(new ForbiddenResponse(MediaException::listingNotAllowed()));
+
+                return;
+            }
+            $medias = $this->readMediaRepository->findByRequestParameters($this->requestParameters);
+            $presenter->presentResponse($this->createResponse($medias));
+        } catch (RequestParametersTranslatorException $ex) {
+            $presenter->presentResponse(new ErrorResponse($ex->getMessage()));
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+        } catch (\Throwable $ex) {
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+            $presenter->presentResponse(new ErrorResponse(MediaException::errorWhileSearchingForMedias()));
+        }
+    }
+
+    private function canAccessToListing(): bool
+    {
+        return $this->user->hasTopologyRole(Contact::ROLE_CONFIGURATION_HOSTS_READ)
+            || $this->user->hasTopologyRole(Contact::ROLE_CONFIGURATION_HOSTS_WRITE);
+    }
+
+    /**
+     * @param \Iterator<int, Media>&\Countable $medias
+     *
+     * @return FindMediasResponse
+     */
+    private function createResponse(\Iterator&\Countable $medias): FindMediasResponse
+    {
+        $response = new FindMediasResponse();
+        /** @var list<Media> $medias */
```

Comment:
Is this phpdoc mandatory ?

Reply:
Just for IDE

Reply:
> potential hidden bugs.

No, because the type is known, but as it was an annotation used during dev, I can delete it.